SpatialGDK/Source/SpatialGDK/Private/Tests/LoadBalanceEnforcerTest.cpp

```diff
@@ -0,0 +1,358 @@
+
+	return VirtualWorkerTranslator;
+}
+
+SpatialGDK::ComponentData MakeComponentDataFromData(Worker_ComponentData Data)
+{
+	return SpatialGDK::ComponentData(SpatialGDK::OwningComponentDataPtr(Data.schema_type), Data.component_id);
+}
+
+SpatialGDK::ComponentUpdate MakeComponentUpdateFromUpdate(Worker_ComponentUpdate Update)
+{
+	return SpatialGDK::ComponentUpdate(SpatialGDK::OwningComponentUpdatePtr(Update.schema_type), Update.component_id);
+}
+
+// Adds an entity to the view with correct LB components. Also assigns the current authority to the passed worker,
+// and the auth intent to the passed virtual worker. The ACL is always assigned to "UnrealWorker".
+// Optionally pass a client name to designate the entity as net owned by that client.
+void AddLBEntityToView(
+	SpatialGDK::EntityView& View, const Worker_EntityId EntityId, const WorkerRequirementSet AuthRequirementSet,
+	const VirtualWorkerId IntentWorkerId,
+	const SpatialGDK::TSchemaOption<PhysicalWorkerName> ClientWorkerName = SpatialGDK::TSchemaOption<PhysicalWorkerName>())
+{
+	AddEntityToView(View, EntityId);
+
+	const WorkerRequirementSet ReadAcl;
+	WriteAclMap WriteAcl;
+	WriteAcl.Add(SpatialConstants::ENTITY_ACL_COMPONENT_ID, UnrealRequirementSet);
+	WriteAcl.Add(SpatialConstants::AUTHORITY_INTENT_COMPONENT_ID, AuthRequirementSet);
+	WriteAcl.Add(SpatialConstants::COMPONENT_PRESENCE_COMPONENT_ID, AuthRequirementSet);
+	WriteAcl.Add(SpatialConstants::NET_OWNING_CLIENT_WORKER_COMPONENT_ID, AuthRequirementSet);
+
+	if (ClientWorkerName.IsSet())
+	{
+		const WorkerRequirementSet AuthClientRequirementSet = WorkerRequirementSet{ WorkerRequirementSet{
+			WorkerAttributeSet{ FString::Printf(TEXT("workerId:%s"), *ClientWorkerName.GetValue()) } } };
+		WriteAcl.Add(SpatialConstants::CLIENT_ENDPOINT_COMPONENT_ID, AuthClientRequirementSet);
+		WriteAcl.Add(SpatialConstants::HEARTBEAT_COMPONENT_ID, AuthClientRequirementSet);
+	}
+
+	AddComponentToView(View, EntityId, MakeComponentDataFromData(SpatialGDK::EntityAcl(ReadAcl, WriteAcl).CreateEntityAclData()));
+	AddComponentToView(View, EntityId, MakeComponentDataFromData(SpatialGDK::AuthorityIntent::CreateAuthorityIntentData(IntentWorkerId)));
+	AddComponentToView(View, EntityId,
+					   MakeComponentDataFromData(SpatialGDK::ComponentPresence::CreateComponentPresenceData(PresentComponents)));
+	AddComponentToView(View, EntityId,
+					   MakeComponentDataFromData(SpatialGDK::NetOwningClientWorker::CreateNetOwningClientWorkerData(ClientWorkerName)));
+	AddComponentToView(View, EntityId, SpatialGDK::ComponentData(SpatialConstants::CLIENT_ENDPOINT_COMPONENT_ID));
+	AddComponentToView(View, EntityId, SpatialGDK::ComponentData(SpatialConstants::HEARTBEAT_COMPONENT_ID));
+	AddComponentToView(View, EntityId, SpatialGDK::ComponentData(SpatialConstants::LB_TAG_COMPONENT_ID));
+
+	AddAuthorityToView(View, EntityId, SpatialConstants::ENTITY_ACL_COMPONENT_ID);
+}
+
+WriteAclMap GetWriteAclMapFromUpdate(const SpatialGDK::EntityComponentUpdate& Update)
```

Comment:
copied this code from the standard library. I was going to make the code in the standard library reusable, but a naive approach would have introduced allocations. This works, and will be going away soon anyway.